main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Shape interface {
+	Area() float64
+	Perimeter() float64
+}
+
+type Circle struct {
+	radius float64
+}
+
+type Rectangle struct {
+	height float64
+	width  float64
+}
+
+func (c Circle) Area() float64 {
+	return math.Pi * c.radius * c.radius
+}
+
+func (c Circle) Perimeter() float64 {
+	return 2 * math.Pi * c.radius
+}
+
+func (r Rectangle) Area() float64 {
+	return r.width * r.height
+}
+
+func (r Rectangle) Perimeter() float64 {
+	return 2 * (r.height + r.width)
+}
+
+func DescribeShape(s Shape) {
+	fmt.Println(s)
```

Comment:
You should create `String() string` method for each struct to have an ability to describe them. For example:
```
func (c Circle) String() string {
	return fmt.Sprintf("This is a circle. Its radius is %.2f ", c.radius)
}
```
After this, `fmt.Println(s)` will print a description you created in `String() string` method